### PR TITLE
SN1-232: fixing organic scoring

### PR DESCRIPTION
**Pull request by Hollyqui**

### CODE REVIEW

File: prompting/organic/organic_loop.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+):
+    logger.debug("[ORGANIC] Waiting for responses to be collected")
+    for _ in range(settings.ORGANIC_TIMEOUT):
+        logger.debug(f"[ORGANIC] {len(completions)} responses collected")
+
+        if len(completions) == len(uids):
+            break
+        await asyncio.sleep(1)
+    else:
+        logger.error("[ORGANIC] Some responses couldn't be collected in time...")
```
Comment (dbobrenko) [inline prompting/organic/organic_loop.py:79]: asyncio.wait_for would be more readable
```suggestion
    async def wait_for_responses():
        while len(completions) < len(uids):
            await asyncio.sleep(0.1)

    try:
        await asyncio.wait_for(wait_for_responses(), timeout=settings.ORGANIC_TIMEOUT)
        logger.debug(f"[ORGANIC] All {len(uids)} responses collected successfully.")
    except asyncio.TimeoutError:
        logger.error("[ORGANIC] Some responses couldn't be collected in time...")
```

File: prompting/organic/organic_loop.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        SynapseStreamResult(
+            accumulated_chunks=completion.accumulated_chunks,
+            accumulated_chunks_timings=(completion.accumulated_chunks_timings),
+            synapse=synapse,
+            uid=completion.uid,
+        )
+        for completion in completions
+    ]
+    logger.debug(f"[ORGANIC] Number of responses collected: {len(completions)}")
+    # return streaming_response
```
Comment (dbobrenko) [inline prompting/organic/organic_loop.py:95]:
```suggestion
```

File: prompting/organic/organic_loop.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+from bittensor.dendrite import dendrite
+from prompting.rewards.scoring import task_scorer
+from prompting.datasets.base import ChatEntry
+from prompting.utils.uids import get_random_uids
+from prompting.base.forward import SynapseStreamResult
+from prompting.base.dendrite import DendriteResponseEvent
+from functools import partial
+
+
+class Completion:
```
Comment (dbobrenko) [inline prompting/organic/organic_loop.py:20]: It should be dataclass

File: prompting/base/validator.py [hunk 4]
```diff
@@ -304,7 +304,7 @@ def update_scores(self, rewards: np.ndarray, uids: list[int]):
         # Compute forward pass rewards, assumes uids are mutually exclusive.
         # shape: [ metagraph.n ]
         step_rewards = np.copy(self.scores)
-        step_rewards[uids] = rewards
+        step_rewards[np.array(uids).astype(int)] = rewards
```
Comment (dbobrenko) [inline prompting/base/validator.py:307]: Why do we need explicit casting?

File: neurons/validator.py [hunk 5]
```diff
@@ -138,6 +104,46 @@ async def run_step(self, k: int, timeout: float) -> ValidatorLoggingEvent | Erro
                 error=str(ex),
             )
 
+    async def collect_responses(self, task: BaseTextTask):
```
Comment (dbobrenko) [inline neurons/validator.py:107]:
```suggestion
    async def collect_responses(self, task: BaseTextTask) -> DendriteResponseEvent:
```

File: neurons/validator.py [hunk 6]
```diff
@@ -138,6 +104,46 @@ async def run_step(self, k: int, timeout: float) -> ValidatorLoggingEvent | Erro
...
+            timeout=settings.NEURON_TIMEOUT,
+            deserialize=False,
+            streaming=True,
+        )
+
+        # Prepare the task for handling stream responses
+        stream_results = await handle_response(stream_results_dict=dict(zip(uids, streams_responses)))
+        logger.debug(
+            f"Non-empty: {len([r.completion for r in stream_results if len(r.completion) > 0])}\n"
+            f"Empty: {len([r.completion for r in stream_results if len(r.completion) == 0])}"
```
Comment (dbobrenko) [inline neurons/validator.py:136]: Add more clarity
```suggestion
            f"Non-empty responses: {len([r.completion for r in stream_results if len(r.completion) > 0])}\n"
            f"Empty responses: {len([r.completion for r in stream_results if len(r.completion) == 0])}"
```

File: tests/organic_queries/check_organic_queries.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
+# ruff: noqa: E402
+import bittensor as bt
```
Comment (dbobrenko) [inline tests/organic_queries/check_organic_queries.py:2]: Will this test be able to run in the CI/CD pipeline?

**Review by dbobrenko — APPROVED** [review]
LGTM with minor comments

File: prompting/base/validator.py (hunk 4 above)
Comment (Hollyqui) [reply prompting/base/validator.py:307]: there was some place where it had the uids as floats... obviously we should fix it in the place where it happens, but with us wanting to release today I didn't have the time to figure out where it is

File: tests/organic_queries/check_organic_queries.py (hunk 7 above)
Comment (Hollyqui) [reply tests/organic_queries/check_organic_queries.py:2]: No, testing files that are run by pytest need to start with test_. So this is a test to run manually

**Pull request merged**